netbird: 0.49.0 -> 0.54.0 + split up + relicense

nazarewk opened the pull request:
## Things done

1. Updating NetBird after brief incompatibilities in `0.52.x` were addressed in `0.53.0`.
2. splitting up packages caused by relicensing of server components
3. adjusted testing flow

Superseeds:
- https://github.com/NixOS/nixpkgs/pull/422784
- https://github.com/NixOS/nixpkgs/pull/423861

- Built on platform:
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [x] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [x] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [x] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [x] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [x] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

nazarewk commented:
Forgot that we've changed server components to AGPLv3 in `0.53.0`, I guess we should at least split server and client-side packages, possibly just 1 package per component.

drupol reviewed (approved):
Really clean!

drupol reviewed (changes requested):
Please fetch the patch from upstream (https://github.com/netbirdio/netbird/pull/4309) instead of hosting a copy of the patch in `nixpkgs`.

phaer commented on pkgs/by-name/ne/netbird/profilemanager-panic-fix.patch:
```diff
@@ -0,0 +1,84 @@
+ 	serviceEnvDesc := `Sets extra environment variables for the service. ` +
+ 		`You can specify a comma-separated list of KEY=VALUE pairs. ` +
+-		`E.g. --service-env LOG_LEVEL=debug,CUSTOM_VAR=value`
++		`E.g. --service-env NB_LOG_LEVEL=debug,CUSTOM_VAR=value`
```
Is that a mistake, or are you renaming the LOG_LEVEL variable deliberately? It doesn't seem related to the patches purpose on first glance?

nazarewk replied:
it indeed doesn't, must have commited the cosmetic change with the fix

phaer commented on pkgs/by-name/ne/netbird/package.nix:
```diff
@@ -73,35 +110,38 @@ buildGoModule (finalAttrs: {
+    let
+      builtBinaryName = lib.last (lib.splitString "/" component.module);
+    in
+    # rename binaries
```
nit: I think it's obvious what it does. Either include an example or remove comments like these :)

phaer reviewed (approved):
Very nice, thanks! I think the patch thing mentioned is technically optional (https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#fetching-patches says "should") but would nevertheless be a good idea. 

See inline feedback, otherwise LGTM!

SuperSandro2000 commented on doc/release-notes/rl-2511.section.md:
```diff
@@ -59,6 +59,8 @@
 
 - `telegram-desktop` packages now uses `Telegram` for its binary. The previous name was `telegram-desktop`. This is due to [an upstream decision](https://github.com/telegramdesktop/tdesktop/commit/56ff5808a3d766f892bc3c3305afb106b629ef6f) to make the name consistent with other platforms.
 
+- `netbird-management` & `netbird-signal` packages got split out of `netbird` due to relicensing of server components to AGPLv3 with version `0.53.0`,
```
Shouldn't this be in the NixOS changelog?

nazarewk replied:
Could you clarify? I'm not 100% sure what you mean.

SuperSandro2000 commented on doc/release-notes/rl-2511.section.md:
```diff
@@ -94,6 +96,7 @@
 
 - `versionCheckHook`: Packages that previously relied solely on `pname` to locate the program used to version check, but have a differing `meta.mainProgram` entry, might now fail.
 
+- added `netbird-upload` & `netbird-relay` packages,
```
Is that even note worthy?

nazarewk replied:
i'll drop it then

nazarewk committed:
netbird: 0.49.0 -> 0.54.0 + split up + relicense

- 0.53.0+ relicensed the server components
  - as a result split into per-component packages
- adds missing 2 upload & relay components
- the tested behavior has changed

nazarewk commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 431976`
Commit: `736451547ca69732fafad8bda4555bf857287af9`

---
### `x86_64-linux`
<details>
  <summary>:fast_forward: 2 packages blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
    <li>tests.nixos-functions.nixos-test</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 6 packages built:</summary>
  <ul>
    <li>netbird</li>
    <li>netbird-management</li>
    <li>netbird-relay</li>
    <li>netbird-signal</li>
    <li>netbird-ui</li>
    <li>netbird-upload</li>
  </ul>
</details>

drupol commented on nixos/doc/manual/release-notes/rl-2511.section.md:
```diff
@@ -182,6 +182,9 @@
 - `services.monero` now includes the `environmentFile` option for adding secrets to the Monero daemon config.
 
+- `services.netbird.server` now uses dedicated packages split out due to relicensing of server components to AGPLv3 with version `0.53.0`,
+
```
Nit: Please keep one blank line between each entry

nazarewk replied:
fixed

phaer commented:
@jvanbruegge Relatable. I wasn't aware of either PR but this one got posted to the nixpgks review requests to matrix, where I saw it. 

Upvoting helps to an extend as you can sort by it while browsing PR labels. I do that from time to time. A more structured approach to distribute reviews among reviewers, maintainers, committers, would surely help with nixpkgs growth though. My first PR to nixpkgs was adding a new service, so it didn't even have a risk of breaking anything, but it took >6 months until it got merged.
